### ISSUE TITLE
Exporting Rail 12v Straight and Curved as Parts List

### ISSUE THREAD

**Issue by henriquedesousa** [Lswbanban/BlueBrick#3]
Hello, and first of all many thanks for developing BueBrick.

One suggestion that I would like to suggest but can be _really_ tricky is the ability to export the 12v Rail Straight (3240BC02) and Curved (3241AC01) into their basic parts. 

For example, the 12v Rail Straight (3240BC02) could be divided into:

- 1x (one) Track 12V Conducting Rail Straight (3242 BL code)
- 2x (two) Track Slotted Rail Straight (3228b BL code)
- 2x (two) Track Sleeper Plate 2 x 8 with Cable Grooves (4166 BL code)

Right now I'm exporting manually, as in pen and paper. I could try to take a look at the code as I'm a developer myself but unfortunately cannot mainly because lack of time due a) my day job is incredibly busy, b) 3 kids, one still a baby and c) I'm a Java guy since ever and cannot learn C# in a normal timeframe due to a) and b).

If you feel it's not worth, feel free to close the issue, no worries.

Again thanks.

**Comment by Lswbanban** [comment]
BlueBrick was not intented to go that deep on the details of the parts. Its focus is mainly do layout planing. I assume you can automatically count the sleepers from the tracks, for example, if you count 2 sleepers for each straight tracks (assuming one is shared with the next track), 2 rails, etc, you can have a pretty good idea of each sub part count. You can also do that in an excel sheet, where you would enter the count of each 12V different type of tracks and it would compute the sub element count.

Alternatively, if you want a more accurate count (especially for the sleepers, when some track are not connected), I suggest you to export your BBM file into LDRAW and open it in a LDRAW software, and ask the part count to the LDRAW software. BlueBrick is normally handling the sleepers parts accurately to avoid duplicated sleepers at the same place when exporting in LDRAW.

To summarize, this is not an easy task with the current version of BlueBrick, and I prefer to avoid this feature for now.

**Comment by henriquedesousa** [comment]
Indeed, that was also my thinking. Cheers for replying and you can close the issue if you fancy.
